crates/red_knot_python_semantic/resources/mdtest/unpacking.md

```diff
@@ -472,3 +472,104 @@ def _(arg: tuple[int, str] | Iterable):
     reveal_type(a)  # revealed: int | bytes
     reveal_type(b)  # revealed: str | bytes
 ```
+
+## For statement
```

Comment:
I've intentionally left out certain test cases, refer to the "Question" in PR description.

Reply:
I think the tests you have here look good!